### DISCUSSION TITLE
How to reprogram ASM3x4x firmware?

### DISCUSSION THREAD

**Discussion by dbg-badger** [smx-smx/ASMTool#12] [Q&A]
As I understand, I can dump my device firmware using this ASMTool utility, but I can't update it.  Is there a different Linux project or tool to do that, or do I need a Windows machine and the ASMedia MPTool?

**Comment by c4801725870** [comment]
> As I understand, I can dump my device firmware using this ASMTool utility, but I can't update it. Is there a different Linux project or tool to do that, or do I need a Windows machine and the ASMedia MPTool?

boot from usb, https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/winpe-create-usb-bootable-drive?view=windows-11
